plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -124,4 +124,6 @@ void createMaterializedView(
     void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment);
 
     Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName);
+
+    void registerTable(ConnectorSession session, SchemaTableName tableName, String tableLocation, Optional<String> metadataFileName);
```

Comment:
move it below `Transaction newCreateTableTransaction(`, since these two are related.